core/src/main/scala/kafka/log/LogLoader.scala

```diff
@@ -424,9 +432,11 @@ class LogLoader(
           // we had an invalid message, delete all remaining log
           warn(s"Corruption found in segment ${segment.baseOffset}," +
             s" truncating to offset ${segment.readNextOffset}")
-          removeAndDeleteSegmentsAsync(unflushed.toList)
+          removeAndDeleteSegmentsAsync(unflushedIter.toList)
```

Comment:
This will iterate unflushedIter. If we don't account for that, remaining segments won't be accurate.

Reply:
Good call! Updated.